go/vt/vttablet/tabletmanager/tm_state.go

```diff
@@ -316,6 +319,11 @@ func (ts *tmState) publishStateLocked(ctx context.Context) {
 		return nil
 	})
 	if err != nil {
+		if topo.IsErrType(err, topo.NoNode) { // someone deleted the tablet record under us. Gracefully exit
+			log.Error("Tablet record has disappeared, shutting down")
+			servenv.ExitChan <- syscall.SIGTERM
```

Comment:
Perhaps we should use `SIGUSR1` instead of `SIGTERM`?

Reply:
from https://www.gnu.org/software/libc/manual/html_node/Miscellaneous-Signals.html:

> The SIGUSR1 and SIGUSR2 signals are set aside for you to use any way you want. They’re useful for simple interprocess communication, if you write a signal handler for them in the program that receives the signal.

Which means `SIGUSR1` can mean _anything_, whereas `SIGTERM` has a clear expected behavior, which is to gracefully shotdown. For this reason. let's keep `SIGTERM` as it provides better clarity.

Reply:
SGTM.